src/services/users-services .js

```diff
@@ -0,0 +1,63 @@
+import { collection, getDocs, getDoc, doc, setDoc, updateDoc, deleteDoc } from 'firebase/firestore';
+import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
+import { db } from '../firebase';
+import { filterMembers } from '../shared/helpers';
+
+export async function getAllUsers() {
+  const querySnapshot = await getDocs(collection(db, 'users'));
+  const users = querySnapshot.docs.map((document) => ({ id: document.id, ...document.data() }));
+
+  return filterMembers(users);
+}
+
+export async function findUser(uid) {
+  const docRef = doc(db, 'users', uid);
+  const docSnap = await getDoc(docRef);
+
+  if (docSnap.exists()) {
+    const { role, name } = docSnap.data();
+
+    return { role, name };
+  }
+
+  return false;
+}
+
+export async function getUserData(uid) {
+  const docRef = doc(db, 'users', uid);
+  const docSnap = await getDoc(docRef);
+
+  return docSnap.data();
+}
+
+async function registerUser({ email, password }) {
+  const auth = getAuth();
+  const user = await createUserWithEmailAndPassword(auth, email, password);
```

Comment:
Are you going to handle errors somehow ? You don't use `try catch` in your app at all

Reply:
Added processing errors of all requests.